test/e2e/network/service.go

```diff
@@ -3798,84 +3703,64 @@ func validateEndpointsPorts(c clientset.Interface, namespace, serviceName string
 	return nil
 }
 
-// restartApiserver restarts the kube-apiserver.
-func restartApiserver(namespace string, cs clientset.Interface) error {
-	// TODO: Make it work for all providers.
-	if !framework.ProviderIs("gce", "gke", "aws") {
-		return fmt.Errorf("unsupported provider for RestartApiserver: %s", framework.TestContext.Provider)
-	}
-	if framework.ProviderIs("gce", "aws") {
-		initialRestartCount, err := getApiserverRestartCount(cs)
-		if err != nil {
-			return fmt.Errorf("failed to get apiserver's restart count: %v", err)
-		}
-		if err := sshRestartMaster(); err != nil {
-			return fmt.Errorf("failed to restart apiserver: %v", err)
-		}
-		return waitForApiserverRestarted(cs, initialRestartCount)
-	}
-	// GKE doesn't allow ssh access, so use a same-version master
-	// upgrade to teardown/recreate master.
-	v, err := cs.Discovery().ServerVersion()
+// restartComponent restarts component static pod
+func restartComponent(cs clientset.Interface, cName, ns string, labelSet labels.Set) error {
+	pods, err := getComponentPod(cs, ns, labelSet)
```

Comment:
this is assuming that the apiserver or kubeproxy are pods, and that is not always true

Reply:
however, seems the test was already considering that, right?

Reply:
the `sshRestartMaster()` is not clear about that :/

Reply:
also the `restartKubeProxy` kills a process

Reply:
It seems that the case before thinking the api-server or kubelet as static pod. So they kill the pods to let kubelet restart the component (kube-proxy or apiserver). As you say, there is no doubt that apiserver or kube-proxy can be running as process. But if we want to restart them, who will restart them when we kill them or other way to stop them? I used to running the local-up-cluster.sh as they are process, no will start them when i kill them.  :）

Reply:
yeah, I'm sorry, seems this is not a big win and overcomplicates everything too much

Reply:
An init system will restart them?

Reply:
let's tackle the Disruptive tests in another PR, and have tese tests now working on KIND jobs
```
ginkgo.It("should be able to up and down services", ...)
ginkgo.It("should implement service.kubernetes.io/service-proxy-name", ...)
ginkgo.It("should implement service.kubernetes.io/headless", ...)
```

Reply:
All right. I will take annother PR to see more details about the change:
```
{"msg":"PASSED [sig-network] Services should work after restarting apiserver [Disruptive]","total":1,"completed":1,"skipped":2981,"failed":0}

{"msg":"PASSED [sig-network] Services should work after restarting kube-proxy [Disruptive]","total":1,"completed":1,"skipped":2061,"failed":0}
```